src/swsssdk/dbconnector.py

```diff
@@ -27,90 +67,134 @@ def load_sonic_db_config(sonic_db_file_path=SONIC_DB_CONFIG_FILE):
-    def get_socket(db_name):
+    def get_instance(db_name, namespace=''):
         if SonicDBConfig._sonic_db_config_init == False:
             SonicDBConfig.load_sonic_db_config()
-        SonicDBConfig.db_name_validation(db_name)
-        return SonicDBConfig.get_instance(db_name)["unix_socket_path"]
+        SonicDBConfig.db_name_validation(db_name, namespace)
+        inst_name = SonicDBConfig._sonic_db_config[namespace]["DATABASES"][db_name]["instance"]
+        SonicDBConfig.inst_name_validation(inst_name, namespace)
+        return SonicDBConfig._sonic_db_config[namespace]["INSTANCES"][inst_name]
 
     @staticmethod
-    def get_hostname(db_name):
+    def get_socket(db_name, namespace=''):
         if SonicDBConfig._sonic_db_config_init == False:
             SonicDBConfig.load_sonic_db_config()
-        SonicDBConfig.db_name_validation(db_name)
-        return SonicDBConfig.get_instance(db_name)["hostname"]
+        return SonicDBConfig.get_instance(db_name, namespace)["unix_socket_path"]
 
     @staticmethod
-    def get_port(db_name):
+    def get_hostname(db_name, namespace=''):
         if SonicDBConfig._sonic_db_config_init == False:
             SonicDBConfig.load_sonic_db_config()
-        SonicDBConfig.db_name_validation(db_name)
-        return SonicDBConfig.get_instance(db_name)["port"]
+        return SonicDBConfig.get_instance(db_name, namespace)["hostname"]
 
     @staticmethod
-    def get_dbid(db_name):
+    def get_port(db_name, namespace=''):
         if SonicDBConfig._sonic_db_config_init == False:
             SonicDBConfig.load_sonic_db_config()
-        SonicDBConfig.db_name_validation(db_name)
-        return SonicDBConfig._sonic_db_config["DATABASES"][db_name]["id"]
+        return SonicDBConfig.get_instance(db_name, namespace)["port"]
 
     @staticmethod
-    def get_separator(db_name):
+    def get_dbid(db_name, namespace=''):
         if SonicDBConfig._sonic_db_config_init == False:
             SonicDBConfig.load_sonic_db_config()
-        SonicDBConfig.db_name_validation(db_name)
-        return SonicDBConfig._sonic_db_config["DATABASES"][db_name]["separator"]
+        SonicDBConfig.db_name_validation(db_name, namespace)
+        return SonicDBConfig._sonic_db_config[namespace]["DATABASES"][db_name]["id"]
+
+    @staticmethod
+    def get_separator(db_name, namespace=''):
+        if SonicDBConfig._sonic_db_config_init == False:
+            SonicDBConfig.load_sonic_db_config()
+        SonicDBConfig.db_name_validation(db_name, namespace)
+        return SonicDBConfig._sonic_db_config[namespace]["DATABASES"][db_name]["separator"]
 
 class SonicV2Connector(DBInterface):
-    def __init__(self, use_unix_socket_path=False, **kwargs):
+    def __init__(self, use_unix_socket_path=True, **kwargs):
         super(SonicV2Connector, self).__init__(**kwargs)
         self.use_unix_socket_path = use_unix_socket_path
+        self.namespace = ''
+
+        # Load sonic DB configs
+        self.load_db_configs()
```

Comment:
This will change old behavior.

Reply:
I need this new behaviour where I have to load the **database_global.json** file in addition to the database_config.json file. If we are in the linux host both these files will be present in the "/var/run/redis/sonic-db". 

If we are in a particular namespace ( for eg: namespace 1) we will only have the file database_config.json alone in the directory  /var/run/**redis1**/sonic-db/" which is mapped to --> "/var/run/redis/sonic-db" in the namespace.

So this API load_db_configs() .. will load either database_config.json alone .. or will load both database_global.json file and database_config.json if present.

Reply:
The default behavior of original code should be kept if possible. I understand your 'new behavior', and you could explicitly invoke SonicDBConfig functions in your limited new use cases (new applications) to achieve the 'new behavior'. But do not change old behavior.

---
In reply to: [402693884](https://github.com/Azure/sonic-py-swsssdk/pull/63#discussion_r402693884) [](ancestors = 402693884)

Reply:
Ok Qi, Agreed. I have changed the logic so that we don't explicitly load the db_configs() as before. 

I made it as a part of the namespace validation logic. In this below API, if the namespace is not equal to default '' namespace, this means a user has explicitly asked for an external namespace. In this case we load the global config if it is not loaded.

    @staticmethod
    def namespace_validation(namespace):
        # Load global config if the namespace is an external one.
        if namespace != '' and SonicDBConfig._sonic_db_global_config_init == False:
            SonicDBConfig.load_sonic_global_db_config()

This way I believe I am not touching the old behavior .. as in the old behavior namespace argument defaults to ''